service/storage_service.cc

```diff
@@ -560,7 +577,12 @@ void storage_service::prepare_to_join(std::vector<inet_address> loaded_endpoints
     app_states.emplace(gms::application_state::VIEW_BACKLOG, versioned_value(""));
     app_states.emplace(gms::application_state::SCHEMA, value_factory.schema(schema_version));
     if (restarting_normal_node) {
+        // Order is important: both the streams timestamp and tokens must be known when a node handles our status.
+        // Exception: there might be no streams timestamp proposed by us if we're upgrading from a non-CDC version.
         app_states.emplace(gms::application_state::TOKENS, value_factory.tokens(my_tokens));
+        if (_streams_ts) {
```

Comment:
It is better to avoid conditional application state injection. Send a value to express explicitly that this node does not have a timestamp. The receiver does not need to guess why the CDC_STREAMS_TIMESTAMP application state is not present. E.g., if (!_stream_ts) send CDC_STREAMS_TIMESTAMP as 'NO_TIMESTAMP'.

Reply:
Done, I'm passing std::optional to value_factory